QuickSubmitForm.inc.php

```diff
@@ -317,33 +315,48 @@ function execute(...$functionParams) {
+			$publication = Services::get('publication')->edit($publication, ['sectionId' => (int) $this->getData('sectionId')], $this->_request);
+		}
+
 		if ($this->getData('articleStatus') == 1) {
-			$publication = $this->_submission->getCurrentPublication();
 			$publication->setData('copyrightYear', $this->getData('copyrightYear'));
 			$publication->setData('copyrightHolder', $this->getData('copyrightHolder'), null);
 			$publication->setData('licenseUrl', $this->getData('licenseUrl'));
 			$publication->setData('pages', $this->getData('pages'));
 			$publication->setData('datePublished', $this->getData('datePublished'));
 			$publication->setData('accessStatus', ARTICLE_ACCESS_ISSUE_DEFAULT);
-			$publication->setData('issueId', $this->getData('issueId'));
-			Services::get('publication')->publish($publication);
+			$publication->setData('issueId', (int) $this->getData('issueId'));
+
+			// If other articles in this issue have a custom sequence, put this at the end
+			$otherSubmissionsInSection = Services::get('submission')->getMany([
+				'contextId' => $this->_request->getContext()->getId(),
+				'issueIds' => [$publication->getData('issueId')],
+				'sectionIds' => [$publication->getData('sectionId')],
+			]);
+			if (count($otherSubmissionsInSection)) {
+				$maxSequence = 0;
+				foreach ($otherSubmissionsInSection as $submission) {
+					if ($submission->getCurrentPublication()->getData('seq')) {
+						$maxSequence = max($maxSequence, $submission->getCurrentPublication()->getData('seq'));
+					}
+				}
+				$publication->setData('seq', $maxSequence + 1);
+			}
 
-			/* FIXME: pkp/pkp-lib#5438
-			$publication = $this->_submission->getCurrentPublication();
-			$publishedSubmissionDao = DAORegistry::getDAO('PublishedArticleDAO');
-			$publishedSubmissionDao->resequencePublishedArticles($this->_submission->getSectionId(), $this->_publishedSubmission->getIssueId());
+			$publication = Services::get('publication')->publish($publication);
 
-			// If we're using custom section ordering, and if this is the first
+			// If this submission's issue uses custom section ordering and this is the first
 			// article published in a section, make sure we enter a custom ordering
-			// for it. (Default at the end of the list.)
-			$sectionDao = DAORegistry::getDAO('SectionDAO');
-			if ($sectionDao->customSectionOrderingExists($this->_publishedSubmission->getIssueId())) {
-				if ($sectionDao->getCustomSectionOrder($this->_publishedSubmission->getIssueId(), $this->_submission->getSectionId()) === null) {
-					$sectionDao->insertCustomSectionOrder($this->_publishedSubmission->getIssueId(), $this->_submission->getSectionId(), REALLY_BIG_NUMBER);
-					$sectionDao->resequenceCustomSectionOrders($this->_publishedSubmission->getIssueId());
+			// for that section to place it at the end.
+			if (DAORegistry::getDAO('SectionDAO')->customSectionOrderingExists($publication->getData('issueId'))) {
```

Comment:
Just a heads-up that I'm trying to avoid inline `DAORegistry::get` calls; PHP Scrutinizer gets confused without type annotations like this:
```
$someDao = DAORegistry::get('SomeDAO'); /* @type $someDao SomeDAO */
```

Reply:
Ack! I've been doing the exact opposite -- converting all assignments into inline calls! :man_facepalming: 

I filed a related issue with the Service classes (https://github.com/pkp/pkp-lib/issues/5328). There I thought it would be worth creating methods with documented return types. 

I'm not wedded to that idea, but one of the things I don't like about `$someDao = DAORegistry::get('SomeDAO'); /* @type $someDao SomeDAO */` is that it introduces a small possibility of ambiguity when searching the codebase. If you're looking for every instance of `SomeDAO::someMethod()` and someone misspelled `$someDao` or decided to name it something different for some reason, then you won't find it and it's more likely to introduce refactoring errors.

Again, not wedded to the methods idea. It seems that DAORegistry, like the Service class, is really a container for singletons. The Service class uses the [Pimple](https://github.com/silexphp/Pimple) container, and I'm not sure exactly what's going on with the syntax there (`$container['some'] = function($c) { ... }` along with some kind of automated injection?). But anyway, whatever approach we adopt we should roll it out wherever we're doing singletons.

Reply:
For service classes, I commented on https://github.com/pkp/pkp-lib/issues/5328#issuecomment-603969636 -- short story yes, getter functions seem like the current best solution (though elsewhere we're trying to move away from getters -- sigh). For DAOs, I'd like to consider the longer term future of these classes before we introduce new patterns for them. Let's hash something out on Slack.